cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go

```diff
@@ -0,0 +1,343 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	log "github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+var CreateRequiredFields = []string{"FederationSettingsId", "OrgId", "ExternalGroupName", "RoleAssignments", "ApiKeys.PrivateKey", "ApiKeys.PublicKey"}
+var ReadRequiredFields = []string{"FederationSettingsId", "Id", "OrgId", "ApiKeys.PrivateKey", "ApiKeys.PublicKey"}
+var UpdateRequiredFields = []string{"FederationSettingsId", "OrgId", "Id", "ExternalGroupName", "RoleAssignments", "ApiKeys.PrivateKey", "ApiKeys.PublicKey"}
+var DeleteRequiredFields = []string{"FederationSettingsId", "OrgId", "ApiKeys.PrivateKey", "ApiKeys.PublicKey"}
+var ListRequiredFields = []string{"ApiKeys.PrivateKey", "ApiKeys.PublicKey"}
+
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-FederatedSettingsOrgRoleMapping")
+}
+
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	_, _ = log.Debugf("Create() currentModel:%+v", currentModel)
+
+	// Validation
+	modelValidation := validateModel(CreateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = log.Debugf("Create - error: %+v", err)
+		return handler.ProgressEvent{
+			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest,
+			Message:          err.Error(),
+			OperationStatus:  handler.Failed,
+		}, nil
+	}
+
+	federationSettingsID := currentModel.FederationSettingsId
+	orgID := currentModel.OrgId
+
+	if (currentModel.ExternalGroupName) == nil {
+		err := errors.New("error creating federated settings org role mapping: ExternalGroupName should be set when `Export` is set")
```

Comment:
do we need to check this?.We are already validating "ExternalGroupName" through validator module.

var CreateRequiredFields = []string{"FederationSettingsId", "OrgId", "**ExternalGroupName**", "RoleAssignments", "ApiKeys.PrivateKey", "ApiKeys.PublicKey"}